ingestors/email_ingestor/email_ingestor.py

```diff
@@ -49,7 +49,12 @@ def _ingest_email_row(self, row: pd.Series) -> List[Document]:
                 # Easier to treat this at utf-8 since str constructor doesn't support all encodings here:
                 # https://chardet.readthedocs.io/en/latest/supported-encodings.html.
                 encoding = 'utf-8'
-            body_str = str(row['body'], encoding=encoding)
+            try:
+                body_str = str(body_str, encoding=encoding)
+            except UnicodeDecodeError:
+                # If illegal characters are found, we ignore them.
+                # I encountered this issue with some emails that had a mix of encodings.
+                body_str = row['body'].decode(encoding, errors='ignore')
```

Comment:
Issues with processing emails were caused by an illegal unicode characters. This fixes by dropping illegal characters.